tensorflow/core/common_runtime/eager/BUILD

```diff
@@ -266,7 +271,14 @@ cc_library(
             "//tensorflow/core/distributed_runtime/eager:remote_execute_node",
             "//tensorflow/core/distributed_runtime/eager:remote_copy_node",
         ],
-    }),
+    }) + if_mkl(["mkl_eager_op_rewrite"]),
+)
+
+cc_library(
+    name = "mkl_eager_op_rewrite",
+    srcs = if_mkl(["mkl_eager_op_rewrite.cc"]),
```

Comment:
Nit: Since the dependency to this target is already guarded by `if_mkl`, I don't think you need `if_mkl` for `srcs` and `copts` here. TensorFlow won't compile this target if it is not in the dependencies.

Reply:
@penpornk  Hi Penporn, based on your suggestion I made the change. However, I got 
the following build errors

... ng/private-tensorflow/.tf_configure.bazelrc: --define=tf_api_version=2
ERROR: /ec/fm/disks/nrv_algo_home01/gzhuang/Working/private-tensorflow/tensorflow/core/common_runtime/eager/BUILD:276:1: //tensorflow/core/common_runtime/eager:mkl_eager_op_rewrite: expected value of type 'list(label)' for attribute 'srcs' in 'cc_library' rule, but got "mkl_eager_op_rewrite.cc" (string)
ERROR: /ec/fm/disks/nrv_algo_home01/gzhuang/Working/private-tensorflow/tensorflow/core/common_runtime/eager/BUILD:276:1: //tensorflow/core/common_runtime/eager:mkl_eager_op_rewrite: expected value of type 'list(string)' for attribute 'copts' in 'cc_library' rule, but got "-DINTEL_MKL=1" (string)
ERROR: /ec/fm/disks/nrv_algo_home01/gzhuang/Working/private-tensorflow/tensorflow/core/distributed_runtime/eager/BUILD:59:1: Target '//tensorflow/core/common_runtime/eager:context' contains an error and its package is in error and referenced by '//tensorflow/core/distributed_runtime/eager:eager_service_impl'
ERROR: /ec/fm/disks/nrv_algo_home01/gzhuang/Working/private-tensorflow/tensorflow/core/distributed_runtime/eager/BUILD:59:1: Target '//tensorflow/core/common_runtime/eager:eager_operation' contains an error and its package is in error and referenced by '//tensorflow/core/distributed_runtime/eager:eager_service_impl'
ERROR: /ec/fm/disks/nrv_algo_home01/gzhuang/Working/private-tensorflow/tensorflow/core/distributed_runtime/eager/BUILD:59:1: Target '//tensorflow/core/common_runtime/eager:execute' contains an error and its package is in error and referenced by '//tensorflow/core/distributed_runtime/eager:eager_service_impl'
ERROR: /ec/fm/disks/nrv_algo_home01/gzhuang/Working/private-tensorflow/tensorflow/core/distributed_runtime/eager/BUILD:59:1: Target '//tensorflow/core/common_runtime/eager:tensor_handle' contains an error and its package is in error and referenced by '//tensorflow/core/distributed_runtime/eager:eager_service_impl'
ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted: Cannot compute config conditions
INFO: Elapsed time: 6.973s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (218 packages loaded, 6383 targets configured)

Reply:
It sounds like you removed the brackets. Please keep the brackets (`srcs` and `copts` take a list, not a string).
```
srcs = ["mkl_eager_op_rewrite.cc"],
copts = ["-DINTEL_MKL=1"],
```

Reply:
@penpornk  Thanks, I will try that.

Reply:
If it doesn't work, please feel free to leave it unchanged. This is minor. Please focus on addressing @alextp's comments (benchmarking results).

Reply:
Give me 15  minutes and I will check in the changes after final round build validation.